hack/verify-licenses.sh

```diff
@@ -0,0 +1,169 @@
+# Deleting vendor directory to make go-licenses fetch license URLs from go-packages source repository
+git worktree add -f "${KUBE_TEMP}"/tmp_test_licenses/kubernetes HEAD >/dev/null 2>&1 || true
+cd "${KUBE_TEMP}"/tmp_test_licenses/kubernetes && rm -rf vendor
+
+
+# Ensure that we find the binaries we build before anything else.
+export GOBIN="${KUBE_OUTPUT_BINPATH}"
+PATH="${GOBIN}:${PATH}"
+
+
+# Explicitly opt into go modules, even though we're inside a GOPATH directory
+export GO111MODULE=on
+
+
+allowed_licenses=()
+packages_flagged=()
+packages_url_missing=()
+exit_code=0
+
+git remote add licenses https://github.com/kubernetes/kubernetes >/dev/null 2>&1 || true
+
+
+# Install go-licenses
+echo -e '[INFO] Installing go-licenses...'
+pushd "${KUBE_TEMP}" >/dev/null
+    git clone https://github.com/google/go-licenses.git >/dev/null 2>&1
+    cd go-licenses
+    go build -o "${GOPATH}/bin"
+popd >/dev/null
+
+
+# Fetching CNCF Approved List Of Licenses
+# Refer: https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md
+curl -s 'https://spdx.org/licenses/licenses.json' -o "${KUBE_TEMP}"/licenses.json
+
+
+number_of_licenses=$(jq '.licenses | length' "${KUBE_TEMP}"/licenses.json)
+loop_index_length=$(( number_of_licenses - 1 ))
+
+
+echo -e '[INFO] Fetching current list of CNCF approved licenses...'
+for index in $(seq 0 $loop_index_length);
+do
+	licenseID=$(jq ".licenses[$index] .licenseId" "${KUBE_TEMP}"/licenses.json)
+	if [[ $(jq ".licenses[$index] .isDeprecatedLicenseId" "${KUBE_TEMP}"/licenses.json) == false ]]
+	then
+		allowed_licenses+=("${licenseID}")
+        fi	
+done
+
+
+# Scanning go-packages under the project & verifying against the CNCF approved list of licenses
+echo -e '[INFO] Starting license scan on go-packages...'
+go-licenses csv --git_remote "licenses" ./... >> "${KUBE_TEMP}"/licenses.csv 2>/dev/null
+
+
+echo -e 'PACKAGE_NAME  LICENSE_NAME  LICENSE_URL\n' >> "${KUBE_TEMP}"/approved_licenses.dump
+while IFS=, read -r GO_PACKAGE LICENSE_URL LICENSE_NAME
+do
+	FORMATTED_LICENSE_URL=""
+	if [[ " ${allowed_licenses[*]} " == *"${LICENSE_NAME}"* ]];
+	then
+		if [[ "${LICENSE_URL}" == 'Unknown' ]];
+		then
+			if  [[ "${GO_PACKAGE}" != k8s.io/* ]];
+			then
+				echo -e "${GO_PACKAGE}  ${LICENSE_NAME}  ${LICENSE_URL}" >> "${KUBE_TEMP}"/approved_licenses_with_missing_urls.dump
+				packages_url_missing+=("${GO_PACKAGE}")
+			else
+				LICENSE_URL='https://github.com/kubernetes/kubernetes/blob/master/LICENSE'
+				echo -e "${GO_PACKAGE}  ${LICENSE_NAME}  ${LICENSE_URL}" >> "${KUBE_TEMP}"/approved_licenses.dump
+			fi
+		elif curl -Is "${LICENSE_URL}" | head -1 | grep -q 404;
+		then
+			# Check whether the License URL is incorrectly formed
+			# TODO: Remove this workaround check once PR https://github.com/google/go-licenses/pull/110 is merged
+			IFS='/' read -r -a split_license_url <<< ${LICENSE_URL}
+			for part_of_url in "${split_license_url[@]}"
+			do
+				if  [[ ${part_of_url} == "" ]]
```

Comment:
```suggestion
				if  [[ ${part_of_url} == '' ]]
```

Single quotes for static strings